internal/native_image/builder.bzl

```diff
@@ -96,33 +191,31 @@ def assemble_native_build_options(
         format_joined = "-H:Features=%s",
     )
 
-    args.add_joined(
-        ctx.attr.initialize_at_build_time,
-        join_with = ",",
-        format_joined = "--initialize-at-build-time=%s",
-    )
-    args.add_joined(
-        ctx.attr.initialize_at_run_time,
-        join_with = ",",
-        format_joined = "--initialize-at-run-time=%s",
-    )
+    # configure the build optimization mode
+    _configure_native_compiler(ctx, args, c_compiler_path, gvm_toolchain)
 
-    if ctx.attr.reflection_configuration != None:
-        args.add(ctx.file.reflection_configuration, format = "-H:ReflectionConfigurationFiles=%s")
-        direct_inputs.append(ctx.file.reflection_configuration)
+    # configure reflection
+    _configure_reflection(ctx, args, direct_inputs)
 
+    # configure resources
     if ctx.attr.include_resources != None:
         args.add(ctx.attr.include_resources, format = "-H:IncludeResources=%s")
 
-    if ctx.attr.jni_configuration != None:
-        args.add(ctx.file.jni_configuration, format = "-H:JNIConfigurationFiles=%s")
-        direct_inputs.append(ctx.file.jni_configuration)
-        args.add("-H:+JNI")
-
     # if a static build is being performed against hermetic zlib, configure it
     if ctx.attr.static_zlib != None:
         _configure_static_zlib_compile(
             ctx,
             args,
             direct_inputs,
         )
+
+    # add test-related flags, if this is a `testonly` target
+    if ctx.attr.testonly:
+        _configure_native_test_flags(
+            ctx,
+            args,
+        )
+
+    # append extra arguments last
+    for arg in ctx.attr.extra_args:
+        args.add(arg)
```

Comment:
fixes bug where extra args were not last

Reply:
Could be `args.add_all`, but doesn't matter too much